### PR TITLE
Make TextInput contain text information

### PR DESCRIPTION
**Pull request by Leinnan**
Simplifies access to the input field content, solves https://github.com/rparrett/bevy_simple_text_input/issues/5
After change, there is an option to detect text change via queries like `text: Query<&TextStorage, Changed<TextStorage>>,`

### CODE REVIEW

**Comment by rparrett** [comment]
Thanks! This generally looks good. I might have a few tweaks when I take a closer look later.

Any chance you could do the reflect / type registrations in a separate PR? And you can leave out the changelog and `Cargo.toml` changes.

**Comment by Leinnan** [comment]
@rparrett done

File: src/lib.rs [hunk 1]
```diff
@@ -171,14 +179,19 @@ fn keyboard(
                 Some(KeyCode::Return) => {
                     submit_writer.send(TextInputSubmitEvent {
                         entity: input_entity,
-                        value: format!("{}{}", text.sections[0].value, text.sections[2].value),
```
Comment (rparrett) [inline src/lib.rs:174]: I think this is slightly incorrect.

If e.g. `delete` and `enter` arrive on the same frame, the event will not contain the results of the `delete`.

Maybe add a local `submitted` variable and do the actual event submission after the value is updated?

File: src/lib.rs [hunk 2]
```diff
@@ -52,6 +54,7 @@ pub struct TextInputBundle {
     cursor_timer: TextInputCursorTimer,
     text_input: TextInput,
     interaction: Interaction,
+    container: TextStorage,
```
Comment (rparrett) [inline src/lib.rs:57]: `container` should be `storage`.

But what do you think about renaming `TextStorage` to `TextInputValue`?

File: src/lib.rs [hunk 3]
```diff
@@ -87,6 +91,10 @@ pub struct TextInput;
 #[derive(Component)]
 struct TextInputInner;
 
+/// A component for the text containing input.
```
Comment (rparrett) [inline src/lib.rs:94]:
```suggestion
/// A component containing the current value of the text input.
```

**Review by rparrett — CHANGES_REQUESTED** [review]
We need to handle users initializing their bundles with a value as well.

**Comment by Leinnan** [comment]
@rparrett Based on your feedback I took a bit of different approach and reuse existing component instead of adding new one. Also reworked a bit the moment when the event is called, so now it should cover the case that you were describing.

**Comment by rparrett** [comment]
Thanks! I'm going to follow up quickly with some small tweaks -- I don't have a ton of time and want to get a build out for Bevy 0.13 this afternoon.

**Pull request merged**